### PR TITLE
refactor: update lints to embark lints 0.3

### PR DESCRIPTION
**Pull request by DeanBDean**
### Checklist

* [x] I have read the [Contributor Guide](../../CONTRIBUTING.md)
* [x] I have read and agree to the [Code of Conduct](../../CODE_OF_CONDUCT.md)
* [x] I have added a description of my changes and why I'd like them included in the section below

### Description of Changes

Updated lints to 0.3 and made changes to satisfy the lints

### Related Issues

https://github.com/EmbarkStudios/relnotes/issues/3

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -44,7 +44,8 @@ where
     if list.is_none() {
         return Ok(None);
     }
-
+    #[allow(clippy::clippy::map_err_ignore)]
```
Comment (XAMPPRocky) [inline src/config.rs]: Isn't it this?
```suggestion
    #[allow(clippy::map_err_ignore)]
```

**Review by XAMPPRocky — APPROVED** [review]
Thank you for your PR! I just have one comment and then can merge

File: src/config.rs (hunk 1 above)
Comment (DeanBDean) [reply src/config.rs]: D'oh

**Comment by XAMPPRocky** [comment]
Thank you for your PR, and congrats on your first contribution! 🎉

**Pull request merged**